src/torrent.cpp

```diff
@@ -5650,20 +5650,22 @@ namespace {
 
 	namespace
 	{
-		std::vector<boost::uint8_t> fix_priorities(std::vector<int> const& input, file_storage const& fs)
+		std::vector<boost::uint8_t> fix_priorities(std::vector<int> const& input
+			, file_storage const* fs)
 		{
 			std::vector<boost::uint8_t> files(input.begin(), input.end());
+			if (fs && files.size() > fs->num_files()) files.resize(fs->num_files());
```

Comment:
There could a problem here, if `files` vector size is less then a number of files in torrent, we'll fix it as is, and then resize it, filling with 4.
What if some of that tail files are paddings? Resulting vector will have non-zero paddings priorities.
I think line 5668 should be moved here to replace this one.